pkg/apis/storage/types.go

```diff
@@ -337,6 +337,35 @@ type CSIDriverSpec struct {
 	//
 	// +optional
 	StorageCapacity *bool
+
+	// ServiceAccountTokens indicates the csi driver needs pods' service account
+	// tokens it is mounting volume for to do necessary authentication. Kubelet
+	// will pass the tokens in VolumeContext in the CSI NodePublishVolume calls.
+	// The CSI driver should parse and validate the following VolumeConext:
+	// "csi.storage.k8s.io/serviceAccount.tokens": {
+	//   'audience-0': {
+	//     'token': token-0,
+	//     'expiry': expiry-0,
+	//   },
+	//   // If audience is not specified, the output key will be ''
+	//   '': {
+	//     'token': token,
+	//     'expiry': expiry,
+	//   },
+	//   ...
+	// }
+	//
+	// Note: Audience in each ServiceAccountToken should be different and at
+	// most one token can ignore Audience.
+	//
+	// +optional
+	// +listType=set
+	ServiceAccountTokens []ServiceAccountToken
+
+	// RequiresRemount indicates the csi driver wants `NodePublishVolume` being
```

Comment:
Is it required for drivers to enable this if they specify service account tokens?

Reply:
no, it is separate.

Reply:
If they set the token to expire, is this required?

Reply:
the token always expires. we can requires `RequiresRepublish` if `ServiceAccountTokens` is not empty but not vice versa (i.e. user can specify `RequiresRepublish` only). WDYT?

Reply:
Yes I think it would be a good idea to require this if they're using serviceaccount tokens.

Reply:
one thing to mention is that some csi drivers might only need the token once and do not care about whether the token expires later. we probably want to keep the flexibility here cause `RequiresRefresh` incurs extra overhead.

Reply:
Ok I guess if that's a valid use case then we don't need to require it. Can you add a comment to the API that RequiresRefresh is the way for a csi plugin to get the new token after it expires?